zerver/tests/test_decorators.py

```diff
@@ -75,6 +78,29 @@ class Request(object):
         result = get_total(request)
         self.assertEqual(result, 21)
 
+    def test_api_key_only_webhook_view(self):
+
+        @api_key_only_webhook_view
+        def get_user_profile_api_key(request, user_profile):
+            return user_profile.api_key
+
+        class Request(object):
+            REQUEST = {} # type: Dict[str, str]
+            COOKIES = {}
+            META = {'PATH_INFO': ''}
+
+        webhook_bot_email = 'webhook-bot@zulip.com'
+        request = Request()
+
+        request.REQUEST['api_key'] = 'not_existing_api_key'
+        with self.assertRaises(JsonableError):
+            get_user_profile_api_key(request)
+
+        request.REQUEST['api_key'] = get_user_profile_by_email(webhook_bot_email).api_key
+        request.COOKIES['csrftoken'] = get_token(request)
+        self.assertEqual(get_user_profile_api_key(request), get_user_profile_by_email(webhook_bot_email).api_key)
```

Comment:
See for example the recent tests I added to zerver/tests/test_decorators.py and zerver/tests/test_auth_backends.py to get a feel for what I have in mind -- the goal is to verify all of our security rules via automated tests so that we don't break any of these security guarantees in future refactoring.